src/components/v5/frame/NavigationSidebar/NavigationSidebar.tsx

```diff
@@ -76,6 +78,35 @@ const NavigationSidebarContent: FC<NavigationSidebarProps> = ({
     }
   };
 
+  useLayoutEffect(() => {
+    switch (openItemIndex) {
+      case 2: {
+        if (trackedIndex !== openItemIndex) {
+          uiEvents.emit(UIEvent.openMenu, { menu: 'dashboard' });
+          setTrackedIndex(openItemIndex);
+        }
+        break;
+      }
+      case 4: {
+        if (trackedIndex !== openItemIndex) {
+          uiEvents.emit(UIEvent.openMenu, { menu: 'finance' });
+          setTrackedIndex(openItemIndex);
+        }
+        break;
+      }
+      case 5: {
+        if (trackedIndex !== openItemIndex) {
+          uiEvents.emit(UIEvent.openMenu, { menu: 'admin' });
+          setTrackedIndex(openItemIndex);
+        }
+        break;
+      }
+      default:
+        setTrackedIndex(-1);
+        break;
+    }
+  }, [openItemIndex, trackedIndex]);
```

Comment:
TODO: Introduce named `enum` keys, rather than cryptic index numbers

Reply:
This cannot happen as `openItemIndex` must be a `number` since the way it was set up, the hook performs math on the various index numbers